src/features/harborView/forms/berthForm/BerthForm.tsx

```diff
@@ -21,6 +21,23 @@ interface BerthFormProps extends FormProps<Berth> {
   pierOptions: Pier[];
 }
 
+interface BerthWithStringifiedDimensions extends Omit<Berth, 'width' | 'length' | 'depth'> {
+  width?: string;
+  length?: string;
+  depth?: string;
+}
+
+const isNumber = (val: string | undefined): boolean => {
+  if (val === undefined) return true; // required is a separate test
+  const numberRegex = RegExp('^[-]?\\d+([.,]\\d+)?$');
+  return numberRegex.test(val);
+};
+
+const isPositive = (val: string | undefined): boolean => {
```

Comment:
Could we extract all these number operations to separate util module and write unit tests for them?